libs/@guardian/libs/src/literalUnionPredicate/literalUnionPredicate.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Create [type predicates][] that narrow primitives
+ * to [literal][] [type unions][].
+ *
+ * [type predicates]: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
+ * [literal]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types
+ * [type unions]: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types
+ *
+ * @example
+ * import { literalUnionPredicate } from '@guardian/libs';
+ *
+ * const stages = ["PROD", "CODE", "DEV"] as const;
+ * const isStage = createPredicate(stages);
+ *
+ * if(!isStage("NOT_A_STAGE")) throw new Error("Invalid stage")
+ */
+export const literalUnionPredicate =
+	<Literals extends readonly unknown[]>(literals: Literals) =>
```

Comment:
The only valid literals are `string` and `number`.

```suggestion
	<Literals extends ReadOnlyArray<string | number>>(literals: Literals) =>
```